pkg/security/security_profile/activity_tree/activity_tree.go

```diff
@@ -542,24 +602,44 @@ func (at *ActivityTree) findBranch(children *[]*ProcessNode, siblings *[]*Proces
 		// directly exec-ed
 		break
 	}
-	return nil, false
+	return nil, 0, false
 }
 
 // rebaseTree rebases the node identified by "nodeIndexToRebase" in the input "tree" onto a newly created branch made of
 // "branchToInsert" and appended to "treeToRebaseOnto". New nodes will be tagged with the input "generationType".
 // This function returns the top level node, owner of the newly inserted branch that lead to the rebased node
-func (at *ActivityTree) rebaseTree(tree *[]*ProcessNode, treeIndexToRebase int, treeToRebaseOnto *[]*ProcessNode, branchToInsert []*model.ProcessCacheEntry, generationType NodeGenerationType, resolvers *resolvers.Resolvers) *ProcessNode {
+func (at *ActivityTree) rebaseTree2(parent ProcessNodeParent, childIndexToRebase int, newParent ProcessNodeParent, branchToInsert []*model.ProcessCacheEntry, generationType NodeGenerationType, resolvers *resolvers.Resolvers) *ProcessNode {
```

Comment:
Why is it renamed to `rebaseTree2`?

Reply:
Good catch ! Thanks :)
I temporarily renamed the function while working on the refactor, but forgot this one.